README.md
![Woody](woody_github_banner.jpg)

![PullRequest Welcome](https://img.shields.io/badge/PR-welcome-brightgreen.svg?style=flat-square)
![Required WP Version](https://img.shields.io/badge/wordpress->=4.8-blue.svg?style=flat-square)
[![Twitter Follow](https://img.shields.io/twitter/follow/raccourciagency.svg?label=Twitter&style=social)](https://twitter.com/raccourciagency)

---

## :fire: Installation & Usage

### Dependencies

Woody is dependent on 2 paid plugins of the wordpress community. To use woody you must absolutely buy and install :

-   Polylang PRO **version 2.9.0**
-   ACF PRO (Advanced Custom Fields PRO) **version 5.9.1**

These plugins must be uploaded to the **web/app/plugins** directory.

> WARNING: If you are using the Woody PRO version, it's possible to install these dependencies by "composer", but this does not mean you do not have to buy these 2 plugins.

### Configuration

Create a configuration file for each of your sites in "config/sites".<br/>
My first project is called "mywebsite", so I create the file **"config/sites/mywebsite/.env"**.<br/>
Some of these settings must be requested from the Raccourci agency, please contact [support@woody-wordpress.com](mailto:support@woody-wordpress.com)<br/>
Here are the available settings

```yml
DB_NAME='wp_mywebsite'
DB_USER='wp_mywebsite'
DB_PASSWORD='mypassword'
DB_HOST='127.0.0.1:3306'

# Use MEMCACHED_SERVERS or MEMCACHED_HOST/MEMCACHED_PORT
MEMCACHED_SERVERS='127.0.0.1:11211;127.0.0.1:11212'
MEMCACHED_HOST='localhost'
MEMCACHED_PORT='11211'

MONGO_SERVERS='localhost:27017'

WP_ENV='dev'
WP_MEMORY_LIMIT='256M'
WP_MAX_MEMORY_LIMIT='256M'
WP_POST_REVISIONS='3'
WP_HOME='https://www.mywebsite.com'
WP_GIT_REPOSITORY='git@github.com:organization/mywebsite.git'
FORCE_SSL_ADMIN='false'
EMPTY_TRASH_DAYS='30'

AUTH_KEY=''
SECURE_AUTH_KEY=''
LOGGED_IN_KEY=''
NONCE_KEY=''
AUTH_SALT=''
SECURE_AUTH_SALT=''
LOGGED_IN_SALT=''
NONCE_SALT=''

WOODY_MAINTENANCE='false'
WOODY_MAINTENANCE_ADMIN='false'
WOODY_ACCESS_STAGING='false'
WOODY_ACCESS_LOCKED='false'

WOODY_ACF_GOOGLE_MAPS_KEY=''
WOODY_ACF_PRO_KEY='You must purchase an ACF license'
WOODY_ADMIN_EMAIL=''
WOODY_ADMIN_NAME=''
WOODY_API_LOGIN='Provided by Raccourci Agency'
WOODY_API_PASSWORD='Provided by Raccourci Agency'
WOODY_CLOUDFLARE_ENABLE='false'
WOODY_CLOUDFLARE_TOKEN=''
WOODY_CLOUDFLARE_URL=''
WOODY_CLOUDFLARE_ZONE=''
WOODY_DEEPL_KEY=''
WOODY_DEEPL_DOMAIN=''
WOODY_GOOGLE_MAPS_API_KEY='Provided by Raccourci Agency'
WOODY_GOOGLE_MAPS_API_KEY_PROD='Provided by Raccourci Agency'
WOODY_GTM=''
WOODY_IGN_MAPS_API_KEY='Provided by Raccourci Agency'
WOODY_IGN_MAPS_API_KEY_PROD='Provided by Raccourci Agency'
WOODY_IMAGE_WEBP_ENABLE='false'
WOODY_LATITUDE='46.1482363'
WOODY_LONGITUDE='-1.1750544'
WOODY_OPTIONS='Provided by Raccourci Agency'
WOODY_PERMALINK_STRUCTURE='/%postname%/'
WOODY_SMTP_HOSTNAME='Provided by Raccourci Agency'
WOODY_SMTP_PORT='587'
WOODY_SMTP_SECURE='tls'
WOODY_SMTP_SENDER='Provided by Raccourci Agency'
WOODY_SMTP_SENDER_NAME='Provided by Raccourci Agency'
WOODY_SMTP_USERNAME='Provided by Raccourci Agency'
WOODY_SMTP_PASSWORD='Provided by Raccourci Agency'
WOODY_SSO_ADD_URL_TOKEN='Provided by Raccourci Agency'
WOODY_SSO_CLIENT_ID='Provided by Raccourci Agency'
WOODY_SSO_CLIENT_SECRET='Provided by Raccourci Agency'
WOODY_SSO_SECRET_URL='https://connect.studio.raccourci.fr'
WOODY_TIMEZONE='Europe/Paris'
WOODY_TOURISTIC_MAPS_API_KEY='Provided by Raccourci Agency'
WOODY_TOURISTIC_MAPS_API_KEY_PROD='Provided by Raccourci Agency'
WOODY_TWIG_CACHE_DISABLE='false'
WOODY_VARNISH_CACHING_COOKIE=''
WOODY_VARNISH_CACHING_DEBUG='true'
WOODY_VARNISH_CACHING_ENABLE='false'
WOODY_VARNISH_CACHING_IPS='127.0.0.1:80'
WOODY_VARNISH_CACHING_TTL='2592000'
WOODY_VARNISH_CACHING_TTL_FOCUSRANDOM='86400'
WOODY_VARNISH_CACHING_TTL_FOCUSSHEET='43200'
WOODY_VARNISH_CACHING_TTL_LIVEPAGE='900'
WOODY_VARNISH_CACHING_TTL_WEATHERPAGE='21600'
WOODY_VARNISH_CACHING_TTL_HAWWWAI_SHEET='2592000'
WOODY_VARNISH_CACHING_TTL_HAWWWAI_PLAYLIST='43200'
```

First, install Woody Core via the Composer package manager (Version 2):

```bash
composer self-update --2
composer install
```

Run this command to install your site:

```bash
woody deploy:core
woody deploy:site -s mywebsite
```

To reinstall your site from scratch

> WARNING: it's necessary to empty its database before launching this command

```bash
woody deploy:core
woody deploy:site -s mywebsite -o force
```

To execute the command but without the gulp compilation:

```bash
woody deploy:site -s mywebsite -o no-gulp
```

The commands can be combined by doing:

```bash
woody deploy:site -s mywebsite -o force,no-gulp
```

Here is the list of available ordering options:

-   force
-   speed
-   no-gulp
-   no-cache
-   no-twig
-   no-varnish
-   no-cdn
-   no-install
-   no-updb
-   no-acf
-   no-warm
-   no-sso

These two commands produce the same result. "speed" is a shortcut to all these options

```bash
woody deploy:site -s mywebsite -o speed
woody deploy:site -s mywebsite -o no-install,no-updb,no-acf,no-gulp,no-warm,no-varnish,no-cdn,no-sso
```

## :inbox_tray: Updating

If you want to make changes in dependencies, you must install this command beforehand

```bash
composer update --prefer-source
```

Otherwise use this command regularly to update yourself

```bash
composer update
```

## :surfer: Sass & Gulp

Run the following command to launch the "watch" of your files.

```bash
cd /gulp
yarn watch --site mywebsite
```

The following command is used to trigger the build of assets for the production.

```bash
cd /gulp
yarn build --site mywebsite
```

## :construction: Debug PHP

There is also the "rcd" function, which allows you to make your own "print_r".

You can add 2 optional parameters:

```php
rcd($val, $exit = false, $pre = true);
```

Exemple :

```php
$array = ['color' => 'red'];
rcd($array, true);
```

## :pill: use WP-CLI

Activate a plugin

```bash
WP_SITE_KEY=mywebsite wp plugin activate hello
```

Import all the SIT (Tourist information system) records of the destination from API (10000 max for the moment)

```bash
WP_SITE_KEY=mywebsite wp woody:hawwwai warm_cache
WP_SITE_KEY=mywebsite wp woody:process async_start
```

Delete all SIT records from the destination

```bash
WP_SITE_KEY=mywebsite wp woody:hawwwai delete_cache
```

Regenerate the canonicals all SIT records

```bash
WP_SITE_KEY=mywebsite wp woody:hawwwai update_canonicals
WP_SITE_KEY=mywebsite wp woody:hawwwai rsdu
```

Update core translations

```bash
WP_SITE_KEY={{sitekey}} wp language core update
```

Update plugin translations (fr_FR example)

```bash
WP_SITE_KEY={{sitekey}} wp language plugin update --all fr_FR
```
or
```bash
WP_SITE_KEY={{sitekey}} wp language plugin update {{plugin-name}} fr_FR
```


## :recycle: Recommended VSCode extensions

-   Git Graph
-   Git Blame
-   BABA-Git Flow
-   EditorConfig for VS Code
-   Beautify
-   PHP-CS-Fixer
-   PHP Extension Pack
-   Sass
-   TODO Highlighter
-   Todo Tree
-   Trailing SPaces
-   Twig
-   TWIG pack
-   Bracket Pair Colorizer
-   Composer
-   ACF Snippet
-   Change String Case
-   Dash
-   Project Manager
-   Wordpress Snippets

Install php-cs-fixer and configure the VSCode extension to launch in "OnSave" mode

```bash
wget https://cs.symfony.com/download/php-cs-fixer-v3.phar -O php-cs-fixer
sudo chmod a+x php-cs-fixer
sudo mv php-cs-fixer /usr/local/bin/php-cs-fixer
```

## :metal: Contributors

Thank you to all the people who have already contributed to Woody Core !

For future contributors, please read our [Contributor Covenant Code of Conduct](CODE_OF_CONDUCT.md)

Header photo by [John Lee on Unsplash](https://unsplash.com/@john_artifexfilms?utm_medium=referral&utm_campaign=photographer-credit&utm_content=creditBadge)<br/>
[![Header photo by John Lee on Unsplash](https://img.shields.io/badge/John%20Lee-black.svg?style=flat-square&logo=unsplash&logoWidth=10)](https://unsplash.com/@john_artifexfilms?utm_medium=referral&utm_campaign=photographer-credit&utm_content=creditBadge)

## :bookmark: License

Woody Core is open-sourced software licensed under the [GPL2](LICENSE).

## :crown: Sponsoring

Woody is a digital ecosystem co-financed by the Regional Tourism Committee of Brittany for [eBreizh Connexion](http://www.ebreizhconnexion.bzh)

![eBreizh Connexion](logo_ebreizh_connexion.png)


CHANGELOG.md
### 06 février 2022 : Version 1.66.0

-   **[Nouveauté]** Nouveau bloc de mise en avant (543 => 547) + amélioration du 530 (titre du bloc)
-   **[Nouveauté]** Module groupe : nouveau type de composant "Excursion"
-   **[Nouveauté]** Visuel & accroche / diaporama : Autoriser les utilisateurs à activer le son de la vidéo
-   **[Mise à jour]** Amélioration du comportement des mises en avant 323, 319 et 326 (slider et titre du bloc)
-   **[Mise à jour]** Retrait du caractère ©️ dans les noms et urls des médias
-   **[Mise à jour]** Harmonisation et correction des miniatures des mises en pages
-   **[Mise à jour]** Infos stations : Prise en compte des tapis roulants dans la liste des remontées mécaniques
-   **[Mise à jour]** Suppression du module "Infolive" (précédemment remplacé par infos stations)
-   **[Correction]** Scroll horizontal avec certaines configurations de sections
-   **[Correction]** Résolution des vignettes dans les résultats de recherche

---

### 26 janvier 2023 : Version 1.65.0

-   **[Nouveauté]** Nouveaux rôles : rédacteur, rédacteur roadBOOK, contributeur roadBOOK
-   **[Mise à jour]** Gestion des droits pour chacun des rôles
-   **[Mise à jour]** Gestion de l'affichage alterné de la mise en avant 360 (si éléments < 4 )
-   **[Mise à jour]** Mise à jour du plugin Redirection de 5.3.8 vers 5.3.9
-   **[Correction]** Affichage des intitulés de prix dans les extraits de fiches SIT
-   **[Correction]** Mise à jour utilisation et remplacement d'image bloquée

---

### 23 janvier 2023

-   **[Mise à jour]** Gestion des fiches SIT dans les cartes interactives
-   **[Mise à jour]** Mise à jour du plugin ACF de 6.0.5 vers 6.0.7
-   **[Mise à jour]** Mise à jour du plugin Redirection de 5.3.6 vers 5.3.8
-   **[Correction]** Optimisation du comportement responsive par défaut de certaines mise en avant
-   **[Correction]** Infos stations : affichage des km de pistes nordique

### 12 janvier 2023

-   **[Nouveauté]** Mise en avant de profils : bouton envoyer un email avec email crypté (options d'affichage)
-   **[Nouveauté]** Organigramme : gestion des entités juridiques

---

### 06 janvier 2023 : Version 1.64.0

-   **[Nouveauté]** Nouveaux templates de mises en avant (155, 617 => 621)
-   **[Nouveauté]** Nouveau template de socialwall (12)
-   **[Mise à jour]** Image carré pour les fiches SIT dans le moteur de recherche
-   **[Mise à jour]** Optimisation des statistiques de téléchargement des pdf dans analytics
-   **[Mise à jour]** Optimisation du datalayer pour les plans de marquage
-   **[Correction]** Infos stations: Nombre de pistes ouvertes = null lorsqu'aucune piste n'est ouverte
-   **[Correction]** undefined affiché dans les résultats de recherche + erreur 404 si pas d'image
-   **[Correction]** Vignettes de templates erronées
-   **[Correction]** Ordre mobile de certains blocs inversé

---

### 28 décembre 2022

-   **[Nouveauté]** Evolution du module "Publicité" pour programmer des encarts dans les pages
-   **[Mise à jour]** Plan de marquage Analytics plus léger

---

### 22 décembre 2022

-   **[Nouveauté]** Affichage de la légende et du copyright des images à l'ouverture d'une galerie
-   **[Nouveauté]** Nouveaux templates de mises en avant (612 => 616)
-   **[Mise à jour]** Mise à jour du plugin ACF de 6.0.4 vers 6.0.5
-   **[Mise à jour]** Exclusion des profils administrateurs dans les restrictions d'accès

---

### 15 décembre 2022 : Version 1.63.0

-   **[Nouveauté]** Restrictions d'accès aux contenus en backoffice (Menu WP => Restrictions d'accès)
-   **[Nouveauté]** Compatibilité GA4
-   **[Nouveauté]** Nouveaux templates de mises en avant (367, 542)
-   **[Mise à jour]** Réduction du nombre d'évènements analytics
-   **[Mise à jour]** Optimisation du comportement et de la charte graphique du moteur de recherche
-   **[Mise à jour]** Liens vers les traductions du média dans le remplacement d'image
-   **[Correction]** Traitement du caractère "-" dans les titres des Visuels & Accroche
-   **[Correction]** Suppression d'une balise style="" dans le rendu des images
-   **[Correction]** Affichage d'une vidéo dans le Visuel & Accroche 17
-   **[Correction]** Ajout d'un séparateur entre les dates dans les extraits de fiches SIT

---

### 10 décembre 2022

-   **[Mise à jour]** Mise à jour du plugin Redirection de 5.3.5 vers 5.3.6
-   **[Correction]** Gestion des dates des widgets "Mon séjour en Montagne"

---

### 1er décembre 2022

-   **[Nouveauté]** Intégration complète de vos widgets "Mon séjour en Montagne"
-   **[Nouveauté]** Intégration du module "Co-marquage Service Public"

---

### 24 novembre 2022

-   **[Nouveauté]** Bloc "Espacement" disponible dans les groupes d'onglets
-   **[Nouveauté]** Nouveaux templates de blocs pub (06, 07)
-   **[Nouveauté]** Nouveaux templates de mise en avant (366, 540, 541)
-   **[Nouveauté]** Nouveau template de socialwall (11)
-   **[Mise à jour]** Mise à jour de Wordpress 6.1.0 vers 6.1.1
-   **[Mise à jour]** Mise en avant sur carte 01 : raccourcissement de la description
-   **[Mise à jour]** "Paramètres des cookies" plus discret en bas de page
-   **[Correction]** Hauteur du sous-menu "burger" en desktop
-   **[Correction]** Suppression du surtitre dans la pagination du slider des mises en avant 130 et 134
-   **[Correction]** Comportement des blocs de publicité avec pubs multiples dans les sections
-   **[Correction]** Mise en avant 154 : prise en charge du 1ier élément
-   **[Correction]** Mise en avant 419 : gestion du responsive
-   **[Correction]** Grille 301 : rétablissement des marges horizontales entre les éléments

---

### 10 novembre 2022 : Version 1.62.0

-   **[Nouveauté]** Options responsive fonctionnelles pour les blocs "galeries médias" et "titres & icones"
-   **[Mise à jour]** Mise à jour du plugin ACF de 6.0.0 vers 6.0.4
-   **[Mise à jour]** Mise à jour du plugin Polylang de 3.2.7 vers 3.2.8
-   **[Mise à jour]** Mise à jour des traductions Wordpress

config/application.php
<?php

/**
 * Your base production configuration goes in this file. Environment-specific
 * overrides go in their respective config/environments/{{WP_ENV}}.php file.
 *
 * A good default policy is to deviate from the production config as little as
 * possible. Try to define as much of your configuration in this file as you
 * can.
 */

use Roots\WPConfig\Config;

/**
 * Directory containing all of the site's files
 */
define('WP_ROOT_DIR', preg_replace('/releases\/\d*/', 'current', dirname(__DIR__)));
define('WP_WEBROOT_DIR', WP_ROOT_DIR . '/web');
define('WP_VENDOR_DIR', WP_ROOT_DIR . '/vendor');
define('WP_CACHE_DIR', WP_ROOT_DIR . '/cache');

/**
 * Expose global env() function from oscarotero/env
 */
Env::init();
define('WP_SITE_KEY', env('WP_SITE_KEY'));

/**
 * Use Dotenv to set required environment variables and load .env file in root
 */
function array_env($env)
{
    $env = str_replace(array('[', ']', '"', ' '), '', env($env));
    return (empty($env)) ? [] : explode(',', $env);
}

if (file_exists(WP_ROOT_DIR . '/config/sites/' . WP_SITE_KEY . '/.env')) {
    $dotenv = Dotenv\Dotenv::create(WP_ROOT_DIR . '/config/sites/' . WP_SITE_KEY);
    $dotenv->load();
    $dotenv->required(['DB_NAME', 'DB_USER', 'DB_PASSWORD', 'DB_HOST', 'WP_HOME', 'WP_ENV']);
}

if (file_exists(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/config/.env')) {
    $dotenv = Dotenv\Dotenv::create(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/config');
    $dotenv->overload();
}

/**
 * Set up our global environment constant and load its config first
 * Default: prod
 */
define('WP_ENV', env('WP_ENV') ?: 'prod');

if (file_exists(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/config/' . WP_ENV . '/.env')) {
    $dotenv = Dotenv\Dotenv::create(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/config/' . WP_ENV);
    $dotenv->overload();
}

/**
 * HTTPS
 */
Config::define('FORCE_SSL_ADMIN', env('FORCE_SSL_ADMIN') ?: false);
if (Config::get('FORCE_SSL_ADMIN')) {
    $scheme = 'https';
    $_SERVER['HTTP_X_FORWARDED_PROTO'] = $scheme;
    $_SERVER['HTTPS'] = 'on';
} else {
    $scheme = 'http';
    $_SERVER['HTTP_X_FORWARDED_PROTO'] = $scheme;
    $_SERVER['HTTPS'] = 'off';
}
Config::define('WP_SCHEME', $scheme);

/**
 * URLs
 */
if (isset($_SERVER['HTTP_X_FORWARDED_HOST'])) {
    $_SERVER['HTTP_HOST'] = $_SERVER['HTTP_X_FORWARDED_HOST'];
}

if (!empty($_SERVER['HTTP_HOST'])) {
    Config::define('WP_HOME', $scheme . '://' . $_SERVER['HTTP_HOST']);
} else {
    Config::define('WP_HOME', env('WP_HOME'));
    $_SERVER['HTTP_HOST'] = parse_url(Config::get('WP_HOME'), PHP_URL_HOST);
}

if (empty($_SERVER['SERVER_NAME'])) {
    $_SERVER['SERVER_NAME'] = $_SERVER['HTTP_HOST'];
}

if (empty($_SERVER['SERVER_PROTOCOL'])) {
    $_SERVER['SERVER_PROTOCOL'] = 'HTTP/1.1';
}

Config::define('WP_SITEURL', Config::get('WP_HOME') . '/wp');

/**
 * Custom Content Directory
 */
Config::define('CONTENT_DIR', '/app');
Config::define('WP_CONTENT_DIR', WP_WEBROOT_DIR . Config::get('CONTENT_DIR'));
Config::define('WP_CONTENT_URL', Config::get('WP_HOME') . Config::get('CONTENT_DIR'));
Config::define('WP_UPLOAD_DIR', Config::get('WP_CONTENT_DIR') . '/uploads/' . WP_SITE_KEY);
Config::define('WP_UPLOAD_URL', Config::get('WP_CONTENT_URL') . '/uploads/' . WP_SITE_KEY);
Config::define('WP_DIST_DIR', Config::get('WP_CONTENT_DIR') . '/dist/' . WP_SITE_KEY);
Config::define('WP_DIST_URL', Config::get('WP_CONTENT_URL') . '/dist/' . WP_SITE_KEY);
Config::define('WP_PLUGINS_DIR', Config::get('WP_CONTENT_DIR') . '/plugins');
Config::define('WP_PLUGINS_URL', Config::get('WP_CONTENT_URL') . '/plugins');
Config::define('WP_MAINTENANCE_DIR', Config::get('WP_CONTENT_DIR') . '/maintenance');
Config::define('WP_TIMBER_DIR', WP_CACHE_DIR . '/timber');

/**
 * DB settings
 */
Config::define('DB_NAME', env('DB_NAME'));
Config::define('DB_USER', env('DB_USER'));
Config::define('DB_PASSWORD', env('DB_PASSWORD'));
Config::define('DB_HOST', env('DB_HOST') ?: 'localhost');
Config::define('DB_CHARSET', 'utf8mb4');
Config::define('DB_COLLATE', '');
$table_prefix = env('DB_PREFIX') ?: 'wp_';

Config::define('MEMCACHED_HOST', env('MEMCACHED_HOST') ?: '');
Config::define('MEMCACHED_PORT', env('MEMCACHED_PORT') ?: '');
Config::define('MEMCACHED_SERVERS', env('MEMCACHED_SERVERS') ?: '');
Config::define('MONGO_SERVERS', env('MONGO_SERVERS') ?: '');

/**
 * Authentication Unique Keys and Salts
 */
Config::define('AUTH_KEY', env('AUTH_KEY') ?: '');
Config::define('SECURE_AUTH_KEY', env('SECURE_AUTH_KEY') ?: '');
Config::define('LOGGED_IN_KEY', env('LOGGED_IN_KEY') ?: '');
Config::define('NONCE_KEY', env('NONCE_KEY') ?: '');
Config::define('AUTH_SALT', env('AUTH_SALT') ?: '');
Config::define('SECURE_AUTH_SALT', env('SECURE_AUTH_SALT') ?: '');
Config::define('LOGGED_IN_SALT', env('LOGGED_IN_SALT') ?: '');
Config::define('NONCE_SALT', env('NONCE_SALT') ?: '');

/**
 * Raccourci WOODY settings
 */
Config::define('WOODY_ACCESS_STAGING', env('WOODY_ACCESS_STAGING') ?: false);
Config::define('WOODY_ACCESS_LOCKED', env('WOODY_ACCESS_LOCKED') ?: false);

Config::define('WOODY_ACF_GOOGLE_MAPS_KEY', array_env('WOODY_ACF_GOOGLE_MAPS_KEY') ?: '');
Config::define('WOODY_ACF_PRO_KEY', env('WOODY_ACF_PRO_KEY') ?: '');
Config::define('WOODY_API_LOGIN', env('WOODY_API_LOGIN') ?: '');
Config::define('WOODY_API_PASSWORD', env('WOODY_API_PASSWORD') ?: '');
Config::define('WOODY_CLOUDFLARE_ENABLE', env('WOODY_CLOUDFLARE_ENABLE') ?: false);
Config::define('WOODY_CLOUDFLARE_TOKEN', env('WOODY_CLOUDFLARE_TOKEN') ?: '');
Config::define('WOODY_CLOUDFLARE_URL', env('WOODY_CLOUDFLARE_URL') ?: '');
Config::define('WOODY_CLOUDFLARE_ZONE', env('WOODY_CLOUDFLARE_ZONE') ?: '');
Config::define('WOODY_DEEPL_KEY', env('WOODY_DEEPL_KEY') ?: '');
Config::define('WOODY_DEEPL_DOMAIN', env('WOODY_DEEPL_DOMAIN') ?: '');
Config::define('WOODY_GOOGLE_MAPS_API_KEY', array_env('WOODY_GOOGLE_MAPS_API_KEY') ?: []);
Config::define('WOODY_GOOGLE_MAPS_API_KEY_PROD', array_env('WOODY_GOOGLE_MAPS_API_KEY_PROD') ?: []);
Config::define('WOODY_GTM', env('WOODY_GTM') ?: '');
Config::define('WOODY_IGN_MAPS_API_KEY', array_env('WOODY_IGN_MAPS_API_KEY') ?: []);
Config::define('WOODY_IGN_MAPS_API_KEY_PROD', array_env('WOODY_IGN_MAPS_API_KEY_PROD') ?: []);
Config::define('WOODY_IMAGE_WEBP_ENABLE', env('WOODY_IMAGE_WEBP_ENABLE') ?: false);
Config::define('WOODY_LATITUDE', env('WOODY_LATITUDE') ?: null);
Config::define('WOODY_LONGITUDE', env('WOODY_LONGITUDE') ?: null);
Config::define('WOODY_OPTIONS', array_env('WOODY_OPTIONS') ?: []);
Config::define('WOODY_PERMALINK_STRUCTURE', env('WOODY_PERMALINK_STRUCTURE') ?: '/%postname%/');
Config::define('WOODY_SSO_ADD_URL_TOKEN', env('WOODY_SSO_ADD_URL_TOKEN') ?: '');
Config::define('WOODY_SSO_CLIENT_ID', env('WOODY_SSO_CLIENT_ID') ?: '');
Config::define('WOODY_SSO_CLIENT_SECRET', env('WOODY_SSO_CLIENT_SECRET') ?: '');
Config::define('WOODY_SSO_SECRET_URL', env('WOODY_SSO_SECRET_URL') ?: '');
Config::define('WOODY_TIMEZONE', env('WOODY_TIMEZONE') ?: 'Europe/Paris');
Config::define('WOODY_TOURISTIC_MAPS_API_KEY', array_env('WOODY_TOURISTIC_MAPS_API_KEY') ?: []);
Config::define('WOODY_TOURISTIC_MAPS_API_KEY_PROD', array_env('WOODY_TOURISTIC_MAPS_API_KEY_PROD') ?: []);
Config::define('WOODY_TWIG_CACHE_DISABLE', env('WOODY_TWIG_CACHE_DISABLE') ?: false);
Config::define('WOODY_VARNISH_CACHING_COOKIE', env('WOODY_VARNISH_CACHING_COOKIE') ?: '');
Config::define('WOODY_VARNISH_CACHING_DEBUG', env('WOODY_VARNISH_CACHING_DEBUG') ?: false);
Config::define('WOODY_VARNISH_CACHING_ENABLE', env('WOODY_VARNISH_CACHING_ENABLE') ?: false);
Config::define('WOODY_VARNISH_CACHING_IPS', env('WOODY_VARNISH_CACHING_IPS') ?: '');
Config::define('WOODY_VARNISH_CACHING_TTL', env('WOODY_VARNISH_CACHING_TTL') ?: 2_592_000);
Config::define('WOODY_VARNISH_CACHING_TTL_FOCUS', env('WOODY_VARNISH_CACHING_TTL_FOCUS') ?: 604_800);
Config::define('WOODY_VARNISH_CACHING_TTL_FOCUSRANDOM', env('WOODY_VARNISH_CACHING_TTL_FOCUSRANDOM') ?: 86400);
Config::define('WOODY_VARNISH_CACHING_TTL_FOCUSSHEET', env('WOODY_VARNISH_CACHING_TTL_FOCUSSHEET') ?: 10800);
Config::define('WOODY_VARNISH_CACHING_TTL_LIVEPAGE', env('WOODY_VARNISH_CACHING_TTL_LIVEPAGE') ?: 900);
Config::define('WOODY_VARNISH_CACHING_TTL_WEATHERPAGE', env('WOODY_VARNISH_CACHING_TTL_WEATHERPAGE') ?: 21600);
Config::define('WOODY_VARNISH_CACHING_TTL_HAWWWAI_SHEET', env('WOODY_VARNISH_CACHING_TTL_HAWWWAI_SHEET') ?: 2_592_000);
Config::define('WOODY_VARNISH_CACHING_TTL_HAWWWAI_PLAYLIST', env('WOODY_VARNISH_CACHING_TTL_HAWWWAI_PLAYLIST') ?: 43200);
Config::define('WOODY_VARNISH_CACHING_TTL_HAWWWAI_PLAYLIST_FMA', env('WOODY_VARNISH_CACHING_TTL_HAWWWAI_PLAYLIST_FMA') ?: 10800);
Config::define('WOODY_ERP_PARTNERID', env('WOODY_ERP_PARTNERID') ?: 0);
Config::define('WOODY_DESTINATION_CONNECT_KEY', env('WOODY_DESTINATION_CONNECT_KEY') ?: '');

/**
 * RoadBook informations
 */
Config::define('ROADBOOK_GA_ID', env('ROADBOOK_GA_ID') ?: '');
Config::define('ROADBOOK_ACCESS_TOKEN', "7qmvHP$*Jpu')%:S");

/**
 * Hawwwai
 */
Config::define('WOODY_HAWWWAI_SHEET', env('WOODY_HAWWWAI_SHEET') ?: false);

/**
 * SMTP informations
 */
Config::define('WOODY_SMTP_HOSTNAME', env('WOODY_SMTP_HOSTNAME') ?: '');
Config::define('WOODY_SMTP_PORT', env('WOODY_SMTP_PORT') ?: 587);
Config::define('WOODY_SMTP_SECURE', env('WOODY_SMTP_SECURE') ?: 'tls');
Config::define('WOODY_SMTP_USERNAME', env('WOODY_SMTP_USERNAME') ?: '');
Config::define('WOODY_SMTP_PASSWORD', env('WOODY_SMTP_PASSWORD') ?: '');
Config::define('WOODY_SMTP_SENDER', env('WOODY_SMTP_SENDER') ?: '');
Config::define('WOODY_SMTP_SENDER_NAME', env('WOODY_SMTP_SENDER_NAME') ?: '');

/**
 * Custom Settings
 */
Config::define('AUTOMATIC_UPDATER_DISABLED', true);
Config::define('ALLOW_UNFILTERED_UPLOADS', false);
Config::define('DISALLOW_FILE_EDIT', true);
Config::define('DISALLOW_FILE_MODS', true);
Config::define('DISABLE_WP_CRON', true);

Config::define('EMPTY_TRASH_DAYS', env('EMPTY_TRASH_DAYS') ?: 30);
Config::define('WP_ALLOW_REPAIR', true);
Config::define('WP_AUTO_UPDATE_CORE', false);
Config::define('WP_MEMORY_LIMIT', env('WP_MEMORY_LIMIT') ?: '256M');
Config::define('WP_MAX_MEMORY_LIMIT', env('WP_MAX_MEMORY_LIMIT') ?: '256M');
Config::define('WP_CRON_LOCK_TIMEOUT', 60);
Config::define('WP_POST_REVISIONS', env('WP_POST_REVISIONS') ?: 3);
Config::define('WP_CACHE_KEY_SALT', WP_SITE_KEY);

/**
 * Get REVISION for logs
 */
if (file_exists(WP_ROOT_DIR . '/REVISION')) {
    $core_revision = substr(file_get_contents(WP_ROOT_DIR . '/REVISION'), 0, 6);
    Config::define('WOODY_CORE_REVISION', $core_revision);
}

if (file_exists(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/REVISION')) {
    $site_revision = substr(file_get_contents(WP_ROOT_DIR . '/web/app/themes/' . WP_SITE_KEY . '/REVISION'), 0, 6);
    Config::define('WOODY_SITE_REVISION', $site_revision);
}

/**
 * Maintenance Mode
 */
if (file_exists(Config::get('WP_MAINTENANCE_DIR') . '/' . WP_SITE_KEY) || file_exists(Config::get('WP_MAINTENANCE_DIR') . '/all')) {
    Config::define('WOODY_MAINTENANCE', true);
} else {
    Config::define('WOODY_MAINTENANCE', env('WOODY_MAINTENANCE') ?: false);
}
Config::define('WOODY_MAINTENANCE_ADMIN', env('WOODY_MAINTENANCE_ADMIN') ?: false);

/**
 * Debugging Settings
 */
Config::define('ENABLE_LOG_DB', env('ENABLE_LOG_DB') ?: false);
Config::define('ENABLE_LOG_FILES', env('ENABLE_LOG_FILES') ?: false);
Config::define('WP_DEBUG', false);
Config::define('WP_DEBUG_DISPLAY', false);
Config::define('SCRIPT_DEBUG', false);

// Polylang
Config::define('PLL_COOKIE', false);
Config::define('PLL_CACHE_LANGUAGES', true);
Config::define('PLL_CACHE_HOME_URL', true);
Config::apply();

$env_config = __DIR__ . '/environments/' . WP_ENV . '.php';
if (file_exists($env_config)) {
    require_once $env_config;
}

/**
 * Bootstrap WordPress
 */
if (!defined('ABSPATH')) {
    define('ABSPATH', WP_WEBROOT_DIR . '/wp/');
}

/**
 * lock mode
 */
require_once(__DIR__ . '/lock.php');

/**
 * maintenance mode
 */
require_once(__DIR__ . '/maintenance.php');

/**
 * debug functions
 */
require_once(__DIR__ . '/debug.php');

/**
 * output functions
 */
require_once(__DIR__ . '/output.php');
